src/Datasource/ResultSetInterface.php

```diff
@@ -22,6 +22,8 @@
 
 /**
  * Describes how a collection of datasource results should look like
+ *
+ * @template T
```

Comment:
Is T not used in other annotations in this interface?

Reply:
I just chose `T` because it was present in the psalm documentation but we could definitely use something more telling. Maybe something like `ResultItem` ?

Reply:
Or (I guess) template variables should always be prefixed with `T` to make them distinguishable from other types so maybe `TResultItem` ?

Reply:
I think he's asking about the lack of the temple usage anywhere else in the interface rather than the template name chosen.

Reply:
Right, yes. The template is not used in here because that interface is empty.
But we need to define that template in here, otherwise 
```
@method \Cake\Datasource\ResultSetInterface<\AlfredProjects\Model\Entity\ProjectStatisticEntry>
```
won't work.

Reply:
Yea, I was asking if it needs to be used in a method to work properly.

Reply:
btw, I'm not blocking this.

Reply:
> Yea, I was asking if it needs to be used in a method to work properly.

Not that I am aware of.

Reply:
Wouldn't the template by the Entity so should we be implementing CollectionInterface<T>?